src/core/client/ui/components/Message/Message.mdx

```diff
@@ -0,0 +1,27 @@
+---
+name: Message
+menu: UI Kit
+---
+
+import { Playground } from 'docz'
+import Message from './Message'
+import HorizontalGutter from '../HorizontalGutter'
+
+# Message
+
+## Basic Use
+<Playground>
+  <HorizontalGutter>
+    <Message>This is a message</Message>
+    <Message fullWidth>Contrary to popular belief, Lorem Ipsum is not simply random text.</Message>
+  </HorizontalGutter>
+</Playground>
+
+## Usage with icon
+<Playground>
+  <HorizontalGutter>
+    <Message icon="alarm">Edit: 1 min 23 secs Remaining</Message>
```

Comment:
From my experience the API to pass in the icon (`<Message icon="alarm">`) has proven to be inferior. It's inflexible as it does not allow to customize which component to use and it does not allow to customize other props of the icon.

A different way would be to allow passing in the Element itself like `<Message icon={<Icon>alarm</Icon>}>`, which is an improvement over the above API. Some flaws include that layouting props must be handled by the `Message` component e.g. `<Message icon={<Icon>alarm</Icon>} alignIcon="right" />`.  Another downside is when using this pattern it can lead to hardly readable Element trees like `<X propA={<Y propB={<Z propC="u" />} propD="v" />} propE="z" />`.

Lastly there is this API that I personally prefer: `<Message><Icon>alarm</Icon> Edit: 1 min 23 secs Remaining</Message>`. It offers the flexibility to pass in any Icon Element, the Message component doesn't need to contain layouting props like when aligning the icon to the right: `<Message>Edit: 1 min 23 secs Remaining <Icon>alarm</Icon></Message>` and if using this pattern often it never leads to hardly readable element trees because it fits nicely into the `jsx` syntax.

The `Button` Component follows the third API pattern already, how do you feel about following this pattern also in this component?